pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -63,80 +63,154 @@ def __init__(
+            else change_dist_size(
+                dist=self.channel_prior, new_size=len(self.channel_columns)
+            )
+        )
 
+    def build_model(
+        self,
+        X: pd.DataFrame,
+        y: pd.Series,
+        **kwargs,
+    ) -> None:
+        self.generate_and_preprocess_model_data(X, y)
+        if self.model_config is None:
+            model_config = self.default_model_config
         else:
-            fourier_features = None
+            model_config = self.model_config
 
-        with pm.Model(coords=coords) as self.model:
+        with pm.Model(coords=self.model_coords) as self.model:
             channel_data_ = pm.MutableData(
                 name="channel_data",
-                value=channel_data,
+                value=self.X[self.channel_columns].to_numpy(),
                 dims=("date", "channel"),
             )
 
-            target_ = pm.MutableData(name="target", value=target_data, dims="date")
+            target_ = pm.MutableData(
+                name="target",
+                value=self.y,
+                dims="date",
+            )
 
-            intercept = pm.Normal(name="intercept", mu=0, sigma=2)
+            intercept = pm.Normal(
+                name="intercept",
+                mu=model_config["intercept"]["mu"],
+                sigma=model_config["intercept"]["sigma"],
+            )
 
+            """beta_channel = pm.HalfNormal(
+                name="beta_channel",
+                sigma=model_config["beta_channel"]["sigma"],
+                dims=model_config["beta_channel"]["dims"],
+            )  # ? Allow prior depend on channel costs?"""
             channel_prior = self._preprocess_channel_prior()
             beta_channel = self.model.register_rv(
                 rv_var=channel_prior, name="beta_channel", dims="channel"
             )
 
-            alpha = pm.Beta(name="alpha", alpha=1, beta=3, dims="channel")
+            alpha = pm.Beta(
+                name="alpha",
+                alpha=model_config["alpha"]["alpha"],
+                beta=model_config["alpha"]["beta"],
+                dims=model_config["alpha"]["dims"],
+            )
 
-            lam = pm.Gamma(name="lam", alpha=3, beta=1, dims="channel")
+            lam = pm.Gamma(
+                name="lam",
+                alpha=model_config["lam"]["alpha"],
+                beta=model_config["lam"]["beta"],
+                dims=model_config["lam"]["dims"],
+            )
 
-            sigma = pm.HalfNormal(name="sigma", sigma=2)
+            sigma = pm.HalfNormal(name="sigma", sigma=model_config["sigma"]["sigma"])
 
             channel_adstock = pm.Deterministic(
                 name="channel_adstock",
                 var=geometric_adstock(
                     x=channel_data_,
                     alpha=alpha,
-                    l_max=adstock_max_lag,
+                    l_max=kwargs["adstock_max_lag"]
+                    if "adstock_max_lag" in kwargs
+                    else 4,
```

Comment:
Do not set a default value. Use must specify it!

Reply:
max_adstock_lag is now a mandatory input parameter